### PR TITLE
adding arrow-odbc

### PR DESCRIPTION
**Pull request by timkpaine**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/arrow-odbc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+build:
+  number: 0
+  script:
+    - {{ PYTHON }} -m build --no-isolation
+    - {{ PYTHON }} -m pip install dist/*.whl --no-deps
+    - cargo-bundle-licenses
+
+requirements:
+  host:
+    - {{ compiler('rust') }}
```
Comment (xhochy) [inline recipes/arrow-odbc/meta.yaml:20]: Compilers should be in build

**Comment by timkpaine** [comment]
@xhochy sorry to bother you, any thoughts on the failure?

**Comment by pacman82** [comment]
@timkpaine hey, thanks for adding this to conda! Sure, I am busy right now and my conda forge knowledge is somewhat limited, but I am happy to help out if I can.

**Comment by pacman82** [comment]
You seem on the right track of fixing it though ...

It's important that it goes against the system preinstalled unixodbc, and unixodbc is not packaged together with arrow-odbc, for both technical and licensing issues.

**Comment by pacman82** [comment]
🎉

**Comment by timkpaine** [comment]
@conda-forge/help-python all set

**Pull request merged**